src/hotspot/share/utilities/utf8.cpp

```diff
@@ -392,6 +392,63 @@ bool UTF8::is_legal_utf8(const unsigned char* buffer, int length,
   return true;
 }
 
+// Return true if `b` could be the starting byte of an encoded 2,3 or 6
+// byte sequence.
+static bool is_starting_byte(unsigned char b) {
+  return b >= 0xC0 && b <= 0xEF;;
+}
+
+// Takes an incoming buffer that was valid UTF-8, but which has been truncated such that
+// the last encoding may be partial, and return the same buffer with a NUL-terminator
+// inserted such that any partial encoding has gone.
+// Note: if the incoming buffer is already valid then we may still drop the last encoding.
+// To avoid that the caller can choose to check for validity first.
+// The incoming buffer is still expected to be NUL-terminated.
+void UTF8::truncate_to_legal_utf8(unsigned char* buffer, int length) {
```

Comment:
Lets make buffer length size_t and avoid awkward casting

Reply:
No this code uses `int` for length everywhere. Feel free to file a RFE to change it all.